pandas/core/internals/blocks.py

```diff
@@ -608,6 +608,15 @@ def replace(
                 blocks = [blk]
             return blocks
 
+        elif value is None and not self.is_object:
```

Comment:
it maybe that we do need to add additional logic for now but there has been some prior discussion about this issue cc @jbrockmendel

Reply:
Yeah this is not a permanent fix, but this happens for every numeric column no matter what value you replace (not only nan), so we should provide a fix before releasing

Reply:
> so we should provide a fix before releasing

sure. (unfortunate that #45725 was not labelled as a regression (from main) and that the milestone was removed)

This had a simple fix, which I reverted because of inconsistencies in handling `None` in a list-like and a scalar. Long term we need to address those inconsistencies.

any temp fix needs a code comment explaining when/how the code can/should be removed.

Reply:
quickly looking back, I did open a PR (#46443) that attempted to address the fact that sometimes we want the missing value to be strict.

we have 2 weeks to fix, so happy to discuss all options.

Reply:
Agreed, would be great if we can come up with a permanent fix

Reply:
has there been a discussion about the desired behavior?  i could imagine having `Series([1, 2]).replace(2, None)` returning `Series([1, np.nan])`.  I think that might be the right thing to do internal-consistency-wise.

To get that I think we'd use _standardize_fill_value at the top of Block.replace.

Reply:
If you do see.loc[0] this upcasts to object, so imo this would make sense for replace too?

Reply:
```
ser = pd.Series([1, 2])
ser.loc[0] = None
>>> ser.dtype
dtype('float64')
```

Reply:
Ah that's on me, I was expanding, e.g. doing ``ser.loc[2] = None`` which casts to object. This is obviously inconsistent too. Would be ok with float64 and casting to nan then

Reply:
> To get that I think we'd use _standardize_fill_value at the top of Block.replace.

IIUC from previous discussions that is the preferred fix. We then need to either special case (or handle differently) the dispatch from the list-like replace, where for backwards compatibility, the `None` replace value is explicit. 

The reason I reverted the original fix for the RecursionError (on main) is to facilitate the backport fix, where we have less flexibility for changes in behavior. But the intention was that the issue would be fixed before 1.5.

The proper fix for 1.5 may also involve discussion on deprecating the current list-like replace behavior to be consistent with a scalar replace.